src/lib/comms/sol-http-client-impl-curl.c

```diff
@@ -914,3 +967,26 @@ sol_http_client_connection_cancel(struct sol_http_client_connection *pending)
     SOL_INT_CHECK(sol_ptr_vector_remove(&global.connections, pending), < 0);
     destroy_connection(pending);
 }
+
+SOL_API struct sol_http_client_connection *
+sol_http_client_request_with_interface(enum sol_http_method method,
+    const char *base_uri, const struct sol_http_params *params,
+    const struct sol_http_request_interface *interface,
+    const void *data)
+{
+    struct sol_http_request_interface *iface;
+    struct sol_http_client_connection *pending;
+
+    iface = calloc(1, sizeof(*iface));
+    SOL_NULL_CHECK(iface, NULL);
+
+    iface = memcpy(iface, interface, sizeof(*iface));
```

Comment:
Why not use `sol_util_memdup()`, instead of allocating memory, clearing it out, then copying over to the cleared-out block?